metrics/dateUtil.py

```diff
@@ -1,28 +1,10 @@
-from datetime import date
-
 import pandas as pd
 
 def main(df_anon, df_original, params, nb_orig_lines):
-    df = df_anon.copy()
-    df_original_reduced = df_original.copy()
```

Comment:
> You have to keep this line because by not doing a copy, the data frames will be modified (even outside function)

Ok. I didn't know that. Thanks. 

> Hello, can you describe me those changes ?

Yeah, sure! 
First, I keep the code which verifies weeks are all the same. Then because of that, we can retrieve the day_of_week using **_pandas.Series.dt.dayofweek_** .

> abs(df_anon.date.dt.dayofweek - df_original.date.dt.dayofweek)

I calculate the difference between 2 series.  The result is a pandas.Series whose values are from 0 to 6. 

Now, I store **3 - result** to **df_score**. **3** means 3 points (I'll divide later). 

> df_score = 3 - abs(df_anon.date.dt.dayofweek - df_original.date.dt.dayofweek)

Values of df_score is in  [ -3 : 3]

> return df_score.loc[df_score > 0].sum() / 3 / nb_orig_line

Finally, we choose only values that bigger than 0 and do the addition, we divide the result by 3 and number_of_lines.